tools/fpgainfo/events.c

```diff
@@ -0,0 +1,124 @@
+//   may be used to  endorse or promote  products derived  from this  software
+//   without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include <getopt.h>
+
+#include "fpgainfo.h"
+#include "events.h"
+#include "board.h"
+
+void events_help(void)
+{
+	printf("\nPrint event log\n"
+	       "        fpgainfo events [-h]");
+	printf("}\n"
+	       "                -b,--boot              Boot index to use, i.e:\n"
+	       "                                         0 for current boot (default),\n"
+	       "                                         1 for previous boot, etc\n"
+	       "                -c,--count             Number of events to print\n"
+	       "                -a,--all               Print all events\n"
+	       "                -s,--sensors           Print sensor data too\n"
+	       "                -i,--bits              Print bit values too\n"
+	       "                -h,--help           Print this help\n"
+	       "\n");
+}
+
+fpga_result events_filter(fpga_properties *filter, int argc, char *argv[])
+{
+	fpga_result res;
+
+	(void)argc;
+	(void)argv;
+
+	res = fpgaPropertiesSetObjectType(*filter, FPGA_DEVICE);
+	if (res != FPGA_OK)
+		OPAE_MSG("setting type to FPGA_DEVICE");
+
+	return res;
+}
+
+fpga_result events_command(fpga_token *tokens, int num_tokens, int argc,
+			   char *argv[])
+{
+	fpga_result res = FPGA_OK;
+	uint32_t first = 0, last = 1;
+	bool print_sensors = false;
+	bool print_bits = false;
+	int i;
+
+	struct option options[] = {
+		{ "boot",    required_argument, NULL, 'b' },
+		{ "count",   required_argument, NULL, 'c' },
+		{ "all",     no_argument,       NULL, 'a' },
+		{ "sensors", no_argument,       NULL, 's' },
+		{ "bits",    no_argument,       NULL, 'i' },
+		{ "help",    no_argument,       NULL, 'h' },
+		{ 0 },
+	};
+
+	while (true) {
+		int opt = getopt_long(argc, argv, "b:c:asih", options, NULL);
+		if (opt == -1)
+			break;
+
+		switch (opt) {
+		case 'b':
+			first = first + strtoul(optarg, NULL, 0);
+			last = last + strtoul(optarg, NULL, 0);
+			break;
+		case 'c':
+			last = last + strtoul(optarg, NULL, 0);
```

Comment:
Need some safeguards here in case optarg is not a number?

Reply:
I added a helper function to check the conversion.